File: java/src/main/java/com/algorithms/sort/InsertionSort.java
package com.algorithms.sort;

import java.util.Arrays;

public class InsertionSort {

    static int [] arr = {6,8,2,4,5,3,7,1};

    public static void main(String[] args) {
        System.out.println(Arrays.toString(arr));
        sort(arr);
        System.out.println(Arrays.toString(arr));
    }

    public static void sort(int [] arr) {

        int counter = 0;
        int i = 0;
        while(i < arr.length) {
            int j = i + 1;

            while(j < arr.length && j > 0 && arr[j] < arr[j-1] ) {
                counter++;
                System.out.println(Arrays.toString(arr)  + "  >> Counter " + counter   + " >>> " + arr[j]);
                int temp = arr[j];
                arr[j] = arr[j-1];
                arr[j-1] = temp;
                j--;
            }
            i++;
        }

    }


}

File: javascript/src/js/sorting/insertionSort.js


function insertionSort(arr) {

    var i = 0;
    while( i < arr.length) {
        var j = i + 1;
        while( j > 0 && j < arr.length) {
            if(arr[j] < arr[j-1]) {
                var temp = arr[j];
                arr[j] = arr[j-1];
                arr[j-1] = temp; 
            }
            j--;
        }
        i++;
        if(j >= arr.length) {
            break;
        }
        console.log(arr);
    }
return arr;
}

let elements = [6,5,4,3,2,1];
// let elements = [5,3,6,1,8];
// let elements = [1]
let  result = insertionSort(elements);
console.log(result);